stl/inc/xstring

```diff
@@ -3980,6 +3980,25 @@ public:
         }
     }
 
+#if _HAS_CXX23
+    template <class _Operation>
+    constexpr void resize_and_overwrite(_CRT_GUARDOVERFLOW const size_type _Newsize, _Operation _Op) {
+        if (_Mypair._Myval2._Myres < _Newsize) {
+            _Reallocate_grow_by(
+                _Newsize - size(), [](_Elem* const _New_ptr, const _Elem* const _Old_ptr, const size_type _Oldsize) {
+                    _Traits::copy(_New_ptr, _Old_ptr, _Oldsize + 1);
+                });
+        }
+
+        const auto _Resultsize = _STD move(_Op)(data(), _Newsize);

```

Comment:
Naming nitpicks: The existing code isn't completely consistent, but there's a strong preference to separate words with underscores: 79 occurrences of `_Old_size` and 30 occurrences of `_New_size`, versus 0 occurrences of `_Oldsize` and only 6 occurrences of `_Newsize`. Thus I believe this function should use `_Old_size`, `_New_size`, `_Result_size`.